### ISSUE TITLE
Rustc unexpectedly panicked

### ISSUE THREAD

**Issue by zonyitoo** [rust-lang/rust#26875]
Backtrace:

```
../src/libcore/iter.rs:804:35: 804:42 error: internal compiler error: Encountered errors `[FulfillmentError(Obligation(predicate=Binder(TraitPredicate(<http::session::DefaultStream as core::cmp::PartialOrd>)),depth=1),Unimplemented)]` fulfilling during trans
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
thread 'rustc' panicked at 'Box<Any>', ../src/libsyntax/diagnostic.rs:176

stack backtrace:
   1:        0x1075863e5 - sys::backtrace::write::hdb195b4e5770538eVns
   2:        0x10758ec13 - panicking::on_panic::hafe0d4ae730c28a3cGw
   3:        0x10754a9f2 - rt::unwind::begin_unwind_inner::h88050f7423a20a57oow
   4:        0x104e8976f - rt::unwind::begin_unwind::h8532912472661560792
   5:        0x104e89707 - diagnostic::SpanHandler::span_bug::haaf8ccf79acf272fl5A
   6:        0x10435fd19 - session::Session::span_bug::hdbb1862f94459432e1s
   7:        0x103c9a94b - trans::common::fulfill_obligation::h0f47bd02a093d3d50Ur
   8:        0x103c85f81 - trans::meth::trans_method_callee::ha0e5a2304805b987TfJ
   9:        0x103cff763 - trans::callee::trans_call_inner::h2562684944855618340
  10:        0x103cdfbc2 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  11:        0x103cb84d7 - trans::expr::trans_into::hc80d5e59b4643c2a4jA
  12:        0x103c454d0 - trans::controlflow::trans_block::h2cc15a51cdc30b54ncv
  13:        0x103c44244 - trans::base::trans_closure::hdcc639dcd785862cZ0h
  14:        0x103c8ee8c - trans::closure::trans_closure_expr::h9b2d2f24a1ad8ab0GMq
  15:        0x103ca271e - trans::consts::const_expr_unadjusted::h8f9785fbe8908b20QTs
  16:        0x103c53005 - trans::consts::const_expr::h625c22442a5060cdGDs
  17:        0x103ca10ed - trans::consts::get_const_expr_as_global::ha098998f45f7267aeys
  18:        0x103c27455 - trans::expr::trans::h7f856dd9d0a4e0bbkqA
  19:        0x103c7cf22 - trans::callee::trans_args::hf6c3eafcb63221bfuBo
  20:        0x103c833c1 - trans::callee::trans_call_inner::h2187404315176819546
  21:        0x103ce0913 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  22:        0x103cdeb32 - trans::expr::trans_unadjusted::h4be66308ca683c69SXA
  23:        0x103c27903 - trans::expr::trans::h7f856dd9d0a4e0bbkqA
  24:        0x103c7cf22 - trans::callee::trans_args::hf6c3eafcb63221bfuBo
  25:        0x103c84bf9 - trans::callee::trans_call_inner::h481791733626692370
  26:        0x103cdf930 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  27:        0x103cb84d7 - trans::expr::trans_into::hc80d5e59b4643c2a4jA
  28:        0x103c454d0 - trans::controlflow::trans_block::h2cc15a51cdc30b54ncv
  29:        0x103c44244 - trans::base::trans_closure::hdcc639dcd785862cZ0h
  30:        0x103c45a42 - trans::base::trans_fn::h1f0f258022927678Mai
  31:        0x103c81e58 - trans::monomorphize::monomorphic_fn::hd970ed37c27c406aGeK
  32:        0x103c714fe - trans::callee::trans_fn_ref_with_substs::hbd189834aea5a3b1iXn
  33:        0x103d3d486 - trans::meth::emit_vtable_methods::closure.51385
  34:        0x103d3b043 - vec::Vec<T>.FromIterator<T>::from_iter::h6035214163836504514
  35:        0x103d39248 - trans::meth::get_vtable::closure.51331
  36:        0x103d3aa15 - iter::FlatMap<I, U, F>.Iterator::next::h17685510477086192067
  37:        0x103d3abfa - vec::Vec<T>::extend_desugared::h7881007509049978589
  38:        0x103d39d03 - iter::Iterator::collect::h14315061346763469760
  39:        0x103ca7628 - trans::expr::unsized_info::h715449673cb7430bDxA
  40:        0x103ce6a83 - trans::expr::coerce_unsized::h42154fb64305c882qIA
  41:        0x103c2813a - trans::expr::trans::h7f856dd9d0a4e0bbkqA
  42:        0x103cb81e5 - trans::expr::trans_into::hc80d5e59b4643c2a4jA
  43:        0x103c478dc - trans::expr::trans_adt::hc5d1f123412f49f1PYB
  44:        0x103c46ddf - trans::base::trans_named_tuple_constructor::h99949572873a589aeei
  45:        0x103c828b1 - trans::callee::trans_call_inner::h2187404315176819546
  46:        0x103ce0913 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  47:        0x103cb84d7 - trans::expr::trans_into::hc80d5e59b4643c2a4jA
  48:        0x103c454d0 - trans::controlflow::trans_block::h2cc15a51cdc30b54ncv
  49:        0x103c44244 - trans::base::trans_closure::hdcc639dcd785862cZ0h
  50:        0x103c45a42 - trans::base::trans_fn::h1f0f258022927678Mai
  51:        0x103c81d1b - trans::monomorphize::monomorphic_fn::hd970ed37c27c406aGeK
  52:        0x103c714fe - trans::callee::trans_fn_ref_with_substs::hbd189834aea5a3b1iXn
  53:        0x103c86571 - trans::meth::trans_method_callee::ha0e5a2304805b987TfJ
  54:        0x103c83d8b - trans::callee::trans_call_inner::h481791733626692370
  55:        0x103cdf930 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  56:        0x103cdeb32 - trans::expr::trans_unadjusted::h4be66308ca683c69SXA
  57:        0x103c27903 - trans::expr::trans::h7f856dd9d0a4e0bbkqA
  58:        0x103c7cf22 - trans::callee::trans_args::hf6c3eafcb63221bfuBo
  59:        0x103c84bf9 - trans::callee::trans_call_inner::h481791733626692370
  60:        0x103cdf930 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  61:        0x103cdeb32 - trans::expr::trans_unadjusted::h4be66308ca683c69SXA
  62:        0x103c27903 - trans::expr::trans::h7f856dd9d0a4e0bbkqA
  63:        0x103c7cf22 - trans::callee::trans_args::hf6c3eafcb63221bfuBo
  64:        0x103c833c1 - trans::callee::trans_call_inner::h2187404315176819546
  65:        0x103ce0913 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  66:        0x103cdeb32 - trans::expr::trans_unadjusted::h4be66308ca683c69SXA
  67:        0x103c27903 - trans::expr::trans::h7f856dd9d0a4e0bbkqA
  68:        0x103cb803f - trans::expr::trans_to_lvalue::h5a3a7ad65443faca1WA
  69:        0x103d235b7 - trans::_match::trans_match_inner::h80262585b88a43c2bLI
  70:        0x103cdff12 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  71:        0x103cb84d7 - trans::expr::trans_into::hc80d5e59b4643c2a4jA
  72:        0x103d2e410 - trans::_match::mk_binding_alloca::h4144802723925626914
  73:        0x103c337d2 - trans::base::init_local::h0889b80b86142b03ljh
  74:        0x103c451b1 - trans::controlflow::trans_block::h2cc15a51cdc30b54ncv
  75:        0x103cdff52 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  76:        0x103cb84d7 - trans::expr::trans_into::hc80d5e59b4643c2a4jA
  77:        0x103cb7fb6 - trans::controlflow::trans_stmt_semi::hdaa3b6a0657ed3derbv
  78:        0x103c45186 - trans::controlflow::trans_block::h2cc15a51cdc30b54ncv
  79:        0x103c44244 - trans::base::trans_closure::hdcc639dcd785862cZ0h
  80:        0x103c45a42 - trans::base::trans_fn::h1f0f258022927678Mai
  81:        0x103c81d1b - trans::monomorphize::monomorphic_fn::hd970ed37c27c406aGeK
  82:        0x103c714fe - trans::callee::trans_fn_ref_with_substs::hbd189834aea5a3b1iXn
  83:        0x103c86571 - trans::meth::trans_method_callee::ha0e5a2304805b987TfJ
  84:        0x103c83d8b - trans::callee::trans_call_inner::h481791733626692370
  85:        0x103cdf930 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  86:        0x103cdeb32 - trans::expr::trans_unadjusted::h4be66308ca683c69SXA
  87:        0x103c27903 - trans::expr::trans::h7f856dd9d0a4e0bbkqA
  88:        0x103cb803f - trans::expr::trans_to_lvalue::h5a3a7ad65443faca1WA
  89:        0x103d235b7 - trans::_match::trans_match_inner::h80262585b88a43c2bLI
  90:        0x103cdff12 - trans::expr::trans_rvalue_dps_unadjusted::h3c597b7a665fcf0cetB
  91:        0x103cb84d7 - trans::expr::trans_into::hc80d5e59b4643c2a4jA
  92:        0x103d2e410 - trans::_match::mk_binding_alloca::h4144802723925626914
  93:        0x103c337d2 - trans::base::init_local::h0889b80b86142b03ljh
  94:        0x103c451b1 - trans::controlflow::trans_block::h2cc15a51cdc30b54ncv
  95:        0x103c44244 - trans::base::trans_closure::hdcc639dcd785862cZ0h
  96:        0x103c45a42 - trans::base::trans_fn::h1f0f258022927678Mai
  97:        0x103c81d1b - trans::monomorphize::monomorphic_fn::hd970ed37c27c406aGeK
  98:        0x103c714fe - trans::callee::trans_fn_ref_with_substs::hbd189834aea5a3b1iXn
  99:        0x103c70124 - trans::callee::trans_fn_ref::hb9b06523ab574ce3EMn
  100:        0x103c8554d - trans::meth::trans_method_callee::ha0e5a2304805b987TfJ
 ... <frames omitted>

Could not compile `solicit`.
```

Rustc version

```
rustc 1.3.0-nightly (6d71ce536 2015-07-06)
```

**Comment by Aatch** [comment]
Could you provide some code that reproduces the error?

**Comment by zonyitoo** [comment]
@Aatch Please build this project: https://github.com/mlalic/solicit

**Comment by arielb1** [comment]
dupe of #26807?

**Comment by zonyitoo** [comment]
Fixed in new nightly build `rustc 1.3.0-nightly (e6a9be10b 2015-07-07)`
